Build Failure: `libdm`

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

It seems like the CMake configuration in the project’s `CMakeLists.txt` is outdated; the minimum required version declared in the file is too old for the version of CMake being used during the Nix build.

### Steps to reproduce

I just ran `nixos-rebuild boot --flake .# --sudo`.

### Expected behaviour

Successful build.

### Screenshots

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/8750h4x3ngrmq3bw95b5rm4f28cqbyx6-source
source root is source/native/LinuxGlobalMenu
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/h276l6ljnavww5mlvfnaxmlh75xyvhz1-libdbm-headers.patch
patching file CMakeLists.txt
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/share/doc/dbm -DCMAKE_INSTALL_INFODIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/include -DCMAKE_INSTALL_SBINDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/p724dq0lm7ka5s64mqxgq1b1lk3vy1qj-libdbm-251.25410.129
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
```

### Additional context

_No response_

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.49, NixOS, 25.11 (Xantusia), 25.11.20251002.7df7ff7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/store/9v6qa656sq3xc58vkxslqy646p0ajj61-source`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/editors/jetbrains/source/build.nix
+++ b/pkgs/applications/editors/jetbrains/source/build.nix
@@ -102,7 +102,12 @@ let
     inherit src;
     sourceRoot = "${src.name}/native/LinuxGlobalMenu";
     patches = [ ../patches/libdbm-headers.patch ];
-    postPatch = "cp ${libdbusmenu-jb}/lib/libdbusmenu-glib.a libdbusmenu-glib.a";
+    postPatch = ''
+      # Fix the build with CMake 4.
+      substituteInPlace CMakeLists.txt \
+        --replace-fail 'cmake_minimum_required(VERSION 2.6.0)' 'cmake_minimum_required(VERSION 3.10)'
+      cp ${libdbusmenu-jb}/lib/libdbusmenu-glib.a libdbusmenu-glib.a
+    '';
     passthru.patched-libdbusmenu = libdbusmenu-jb;
     installPhase = ''
       runHook preInstall

